tests/annotator/annotator.cpp

```diff
@@ -1598,11 +1622,23 @@ TEST(Annotator, assignAllIdsNonblankModel)
     auto model = parser->parseModel(modelStringUniqueIds);
     auto annotator = libcellml::Annotator::create();
 
-    annotator->assignAllIds(model);
+    EXPECT_FALSE(annotator->assignAllIds(model));
     EXPECT_EQ(size_t(0), annotator->errorCount());
     EXPECT_EQ(modelStringUniqueIds, printer->printModel(model));
 }
 
+TEST(Annotator, assignAllIdsDuplicateIdsModel)
+{
+    auto parser = libcellml::Parser::create();
+    auto printer = libcellml::Printer::create();
+    auto model = parser->parseModel(modelStringLotsOfDuplicateIds);
+    auto annotator = libcellml::Annotator::create();
+
+    EXPECT_FALSE(annotator->assignAllIds(model));
+    EXPECT_EQ(size_t(0), annotator->errorCount());
```

Comment:
if `assignAllIds(model)` goes back to not caching the current model, then this should be non-zero...